### PR TITLE
update black to 22.3.0

### LINKED ISSUE #3893: cannot import name '_unicodefun' from 'click'
I am reporting a problem with Biopython version, Python version, and operating
system as follows:

Python 3.9.12 (main, Mar 26 2022, 15:51:15) 
[Clang 13.1.6 (clang-1316.0.21.2)] on darwin
>>> import sys; print(sys.version)
3.9.12 (main, Mar 26 2022, 15:51:15) 
>>> import platform; print(platform.python_implementation()); print(platform.platform())
CPython
macOS-12.3.1-x86_64-i386-64bit
>>> import Bio; print(Bio.__version__)
1.80.dev0

I'm struggling again with a commit; the pre-commit environment re-initialized and now I have this issue running black:

```
black....................................................................Failed
- hook id: black
- exit code: 1

Traceback (most recent call last):
  File "/Users/rob/.cache/pre-commit/repof1qgx4jh/py_env-python3.9/bin/black", line 8, in <module>
    sys.exit(patched_main())
  File "/Users/rob/.cache/pre-commit/repof1qgx4jh/py_env-python3.9/lib/python3.9/site-packages/black/__init__.py", line 1423, in patched_main
    patch_click()
  File "/Users/rob/.cache/pre-commit/repof1qgx4jh/py_env-python3.9/lib/python3.9/site-packages/black/__init__.py", line 1409, in patch_click
    from click import _unicodefun
ImportError: cannot import name '_unicodefun' from 'click' (/Users/rob/.cache/pre-commit/repof1qgx4jh/py_env-python3.9/lib/python3.9/site-packages/click/__init__.py)

```

I've installed click from PyPI as the error seems to indicate, and have black version 22.3.0.  The same issue has appeared in [data-utils](https://github.com/owid/data-utils-py/issues/10) but the work there does not seem to help me.  I've cleared my ~/.cache/, updated packages, and `pre-commit install`'d repeatedly but to no avail.

Happy if anyone can highlight what I'm missing?

### PR DESCRIPTION
**Pull request by rob-miller**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This tiny PR changes the black dependency to 22.3.0 in .pre-commit-config.yaml and closes #3893, however
I think black now triggers on 

    Bio/SwissProt/__init__.py:699:9: B020 Found for loop that reassigns the iterable it is iterating with each iterable value.
    Bio/Affy/CelFile.py:177:9: B020 Found for loop that reassigns the iterable it is iterating with each iterable value.
    Bio/SCOP/__init__.py:333:13: B020 Found for loop that reassigns the iterable it is iterating with each iterable value.
    Bio/SCOP/__init__.py:339:13: B020 Found for loop that reassigns the iterable it is iterating with each iterable value.
    Bio/SCOP/__init__.py:346:13: B020 Found for loop that reassigns the iterable it is iterating with each iterable value.

but I did not attempt to address these as not familiar with the code - but happy to dive in if no one else is available.

### CODE REVIEW

**Comment by peterjc** [comment]
The pre-commit.ci service will periodically update all these, but does like this example occasionally reveal newly identified issues.

Those five B020 issues should all probably be fixed. They tend to be simple fixes... but might be hidden bugs in our code?

**Comment by peterjc** [comment]
Cross reference #3895

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ repos:
     rev: v1.12.1
     hooks:
     -   id: blacken-docs
-        additional_dependencies: [black==22.1.0]
+        additional_dependencies: [black==22.3.0]
```
[the remark this answers is not in the document]
Comment (rob-miller) [reply .pre-commit-config.yaml:39]: Sorry didn't see the line above.  

Seems better now to let the pre-commit.ci service finish?  Happy if you merge that one and we close this one.

**Comment by peterjc** [comment]
OK, but I'd be happy to get a fix addressing those B020 issues (nice stylistic improvement to fix a code-smell).

**Pull request closed without merging**